TypeCobol/Compiler/Preprocessor/ReplaceTokensLinesIterator.cs

```diff
@@ -264,10 +264,13 @@ public Token NextToken()
                     {
                         if (currentPosition.ReplaceOperations != null)
                         {
-                            foreach (ReplaceOperation replaceOperation in currentPosition.ReplaceOperations)
+                            for(int i = 0; i < currentPosition.ReplaceOperations.Count; i++)
                             {
+                                ReplaceOperation replaceOperation = currentPosition.ReplaceOperations[i];
                                 status = TryAndReplace(nextToken, replaceOperation);
-                                if (status.replacedToken != null) return status.replacedToken;
+                                if (status.replacedToken != null)
+                                    return ApplyRemainingReplaces(i + 1, status, currentPosition.ReplaceOperations);
```

Comment:
Why not let the enclosing loop continue till all replace have been done ?
I mean, what you have done in ApplyRemainingReplaces could be done right here.

Reply:
Not exactly the behavior is slightly different..

Reply:
No there is also the return of the lastReplacedToken

Reply:
Yes we agree that there are slight differences, but still it's possible to factorize it.

Reply:
No, there's a return value here, depending on a continuation.

Reply:
ok done in  041d2e1